File: MAP(Advanced Programming Methods)/lab4/src/Model/ADT/IFileTable.java
package Model.ADT;

public interface IFileTable<K,V> {

    public void add(K key,V val);
    public void remove(K key);
    public boolean contains(K key);
    public V get(K key);
    public Iterable<K> getAll();
    public Iterable<V> getValues();
}

File: MAP(Advanced Programming Methods)/LabAssig7/LabAssig7/Model/ADT/IFileTable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Collections;
namespace LabAssig7.Model.ADT
{
    public interface IFileTable<K,V>:IEnumerable
    {
        void AddValue(K key, V value);
        void UpdateValue(K key, V value);
        void Remove(K key);
        bool Contains(K key);
        V GetValue(K key);
    }
}
